openhtf/plugs/usb/__init__.py

```diff
@@ -36,56 +38,153 @@ def MyPhase(test, adb):
 conf.Declare('usb_server', 'USB Server IP/Hostname')
 conf.Declare('usb_server_port', 'USB Server Port',
              default_value=10000)
-
 conf.Declare('libusb_rsa_key', 'A private key file for use by libusb auth.')
+conf.Declare('device_type', 'android, fastboot or general usb type')
```

Comment:
hmm, this device type specify the android, fastboot or pure usb device. it is true, if it is a android/fastboot/usb device, just use AdbPlug/FastbootPlug/UsbPlug, since internally, AdbPlug/FastbootPlug are both based on the usb handle to do another level connection, I simplified the code base on each plug, and get the info from config to open the underneath usb device.